src/ECS/ECS.cpp

```diff
@@ -0,0 +1,179 @@
+            // for(auto entity: view) {
+            //     auto &vel = view.get<velocity>(entity);
+            //     // ...
+            // }
+        }
+
+        entt::registry initializeECS () {
+            entt::registry registry;
+
+            // for(auto i = 0u; i < 10u; ++i) {
+            //     const auto entity = registry.create();
+            //     registry.emplace<position>(entity, i * 1.f, i * 1.f);
+            //     if(i % 2 == 0) { registry.emplace<velocity>(entity, i * .1f, i * .1f); }
+            // }
+
+            // update(registry);
+
+            Logger::message(System::ECS) << "Initialized ECS" << std::endl;
+
+            return registry;
+        }
+
+        bool isTreeType (EntityType type) {
+            switch (type) {
+                case EntityType::TREE_APPLE:
+                case EntityType::TREE_BIRCH:
+                case EntityType::TREE_CHESTNUT:
+                case EntityType::TREE_OAK:
+                case EntityType::TREE_PINE:
+                case EntityType::TREE_SHRUB1:
+                case EntityType::TREE_SHRUB2:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+        
+        void renderUnit (PositionComponent position, AnimatedComponent animationComponent, std::shared_ptr<Gm1File> assetFile) {
+            Camera& cam = Camera::instance();
+            auto [x, y] = position;
+
+            int px = x * 30 - cam.positionX;
+            int py = y * 15 - cam.positionY;
+
+            SDL_Rect r = assetFile->GetTextureAtlas()->Get(animationComponent.animate ? animationComponent.frameIndex : 0); // TODO: index
+            Rendering::Render(*assetFile->GetTextureAtlas(), px, py, &r);
+        }
+
+        void render (entt::registry &registry) {
+            auto view = registry.view<const EntityTypeComponent, const PositionComponent, const AnimatedComponent>();
+
+            // this is just getting pointers to preloaded, parsed file data
+            // but maybe it can be moved somewhere else if this is performance hit
+            auto lordFile = AssetStorage::getFileGM1(AssetStorageKey::BODY_LORD_GM1);
+            auto deerFile = AssetStorage::getFileGM1(AssetStorageKey::BODY_DEER_GM1);
+
+            auto appleTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_APPLE_GM1);
+            auto birchTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_BIRCH_GM1);
+            auto chestnutTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_CHESTNUT_GM1);
+            auto oakTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_OAK_GM1);
+            auto pineTreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_PINE_GM1);
+            auto shrub1TreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_SHRUB1_GM1);
+            auto shrub2TreeFile = AssetStorage::getFileGM1(AssetStorageKey::TREE_SHRUB2_GM1);
```

Comment:
Can this be const ?

Reply:
I changed a bit where it is and how it loads, but I think the new place can be const.

Reply:
I changed a bit where it is and how it loads, but I think the new place can be const.